tests/issues/issue-268/formatted.pug

```diff
@@ -0,0 +1,46 @@
+doctype html
+html
+	head
+	body
+		#indentation-tests
+			.short(style="display: none") Hello World
+			.getting-longer(style="display: none") I can't just say "Hello World" here
+			.i-am-not-being-creative-with-these-names(
+				style="display: none",
+				color="red"
+			).
+				There is surely more I could write here
+			.short-again(style="display: none", color="red", font="Sans Serif")
+				| I could surely more I could write here
+
+			.hi(data-wrap="false") This should be wrapped
+
+		section#contact
+			.container
+				h3 Contact
+				form(name="contact", method="POST")
+					.row.gtr-uniform
+						.col-6.col-12-x-small
+							input#name(type="text", name="name", placeholder="Name *", required)
+						.col-6.col-12-x-small
+							input#email(type="email", name="email", placeholder="Email *", required)
```

Comment:
@OiYouYeahYou do you mean these two lines?

![image](https://user-images.githubusercontent.com/7195563/128220170-2dbeabae-a772-4540-9ba1-321e918b116e.png)

I see that they are longer than 80 cols :thinking:  
And you want it that they wrap?

Reply:
It differs depending on what indentation width you are using, but yes those two lines are a problem. In my opinion the formatting should wrap identically to the spaces.

The test I've been using replaces tabs for spaces at the same with and then compares them:

```ts
function comparisonTabTest(message: string, { output, tabWidth }: { output: string; tabWidth: number }): void {
	test(message, () => {
		const spaces: string = format(unformatted, { parser: 'pug', plugins: [plugin], tabWidth, useTabs: false });
		const tabs: string = format(unformatted, {
			parser: 'pug',
			plugins: [plugin],
			tabWidth,
			useTabs: true
		}).replace(/\t/g, ' '.repeat(tabWidth));

		expect(tabs).toBe(spaces);
	});
}
```

I've already looked at fixing this and it's non-trivial to repair at the moment. There is a hacky solution though, which is to use `useTabs: false` and then convert the spaces to tabs after printing.

Reply:
Okay now I fully understood the original issue ^^

Yeah, we have a `currentLineLength`

https://github.com/prettier/plugin-pug/blob/998a192e7d1f5ba1af4f98aacf0448dbcc26364a/src/printer.ts#L174

And then we have lines like 

https://github.com/prettier/plugin-pug/blob/998a192e7d1f5ba1af4f98aacf0448dbcc26364a/src/printer.ts#L1000

where we using `indentString.length`

But it only counts as `1` instead of e.g. `8` for each `\t`.
So we could try to create a `get indentStringLength() {...}` and use it instead.

You have write access to this branch `issue-268`. Would you like to try it out?
I don't have the time to code for the next ~40 hours :slightly_frowning_face:

Reply:
That was much more easy than I thought :eyes: